From cc5fa20e86a3b76814abeb8d6726f0f6c3b66d62 Mon Sep 17 00:00:00 2001
From: Miguel Gomez <magomez@igalia.com>
Date: Thu, 31 Jul 2025 23:49:57 -0700
Subject: [PATCH] REGRESSION(297498@main): [WPE] build broken when using API
 1.1 https://bugs.webkit.org/show_bug.cgi?id=296738

Reviewed by Carlos Garcia Campos.

Update the webkitJavascriptResultCreate() function so it receives a
WebKit::JavaScriptEvaluationResult&& parameter instead of a
WebCore::SerializedScriptValue& one.

Also, add a missing JavaScriptEvaluationResult.h include.

* Source/WebKit/UIProcess/API/glib/WebKitJavascriptResult.cpp:
(webkitJavascriptResultCreate):

Canonical link: https://commits.webkit.org/298116@main
---
 Source/WebKit/UIProcess/API/glib/WebKitJavascriptResult.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/Source/WebKit/UIProcess/API/glib/WebKitJavascriptResult.cpp b/Source/WebKit/UIProcess/API/glib/WebKitJavascriptResult.cpp
index cc724bb49be6..c65a360e7c8e 100644
--- a/Source/WebKit/UIProcess/API/glib/WebKitJavascriptResult.cpp
+++ b/Source/WebKit/UIProcess/API/glib/WebKitJavascriptResult.cpp
@@ -23,6 +23,7 @@
 #if !ENABLE(2022_GLIB_API)
 
 #include "APISerializedScriptValue.h"
+#include "JavaScriptEvaluationResult.h"
 #include "WebKitJavascriptResultPrivate.h"
 #include <jsc/JSCContextPrivate.h>
 #include <jsc/JSCValuePrivate.h>
@@ -46,10 +47,10 @@ struct _WebKitJavascriptResult {
 
 G_DEFINE_BOXED_TYPE(WebKitJavascriptResult, webkit_javascript_result, webkit_javascript_result_ref, webkit_javascript_result_unref)
 
-WebKitJavascriptResult* webkitJavascriptResultCreate(WebCore::SerializedScriptValue& serializedScriptValue)
+WebKitJavascriptResult* webkitJavascriptResultCreate(WebKit::JavaScriptEvaluationResult&& scriptResult)
 {
     WebKitJavascriptResult* result = static_cast<WebKitJavascriptResult*>(fastMalloc(sizeof(WebKitJavascriptResult)));
-    new (result) WebKitJavascriptResult(serializedScriptValue);
+    new (result) WebKitJavascriptResult(WTFMove(scriptResult));
     return result;
 }
 
